### ISSUE TITLE
[Bug] __timeFilter() does not support MySQL functions

### ISSUE THREAD

**Issue by brunomagalhaes** [grafana/grafana#8728]
Hi everyone,

First of all, I'm LOVING Grafana and it's multiple data sources! I have a InfluxDB collecting all our DigitalOcean droplets metrics and now we are thinking about replacing our in-house BI dashboards with Grafana as well, and we would like to contribute as much as we can to this amazing platform.

Now, what we discovered is that __timeFilter() is messing with MySQL functions, let me explain:

1) We have the following table to store article acesses (and many more like this for other objects types):
```
CREATE TABLE `tb_content_articles_accesses` (
  `article_id` int(20) unsigned NOT NULL DEFAULT '0',
  `accesses_date` date NOT NULL DEFAULT '0000-00-00',
  `accesses_time` time NOT NULL DEFAULT '00:00:00',
  `accesses_total` int(20) unsigned NOT NULL DEFAULT '0',
  PRIMARY KEY (`article_id`,`accesses_date`,`accesses_time`),
  KEY `dc_content_articles_accesses_a` (`accesses_time`),
  KEY `dc_content_articles_accesses_b` (`article_id`,`accesses_time`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 PACK_KEYS=1 ROW_FORMAT=COMPACT;
```

2) Then we are trying to create the following dashboard (screenshot) using the MySQL datasource:
```
SELECT
  UNIX_TIMESTAMP(CONCAT(accessess_date, ' ', accessess_time)) as time_sec,
  SUM(accessess_total) as value,
  article_id as metric
FROM tb_content_articles_accessess
WHERE $__timeFilter(CONCAT(accessess_date, ' ', accessess_time)) 
ORDER BY CONCAT(accessess_date, ' ', accessess_time) ASC
```
3) That gets incorrectly translated to:
```
SELECT
  UNIX_TIMESTAMP(CONCAT(accesses_date, ' ', accesses_time)) as time_sec,
  SUM(accesses_total) as value,
  article_id as metric
FROM tb_content_articles_accesses
WHERE CONCAT(accesses_date, ' ', accesses_time > FROM_UNIXTIME(1498626836) AND CONCAT(accesses_date, ' ', accesses_time < FROM_UNIXTIME(1498648436))
ORDER BY accesses_date ASC, accesses_time ASC
```

Please note that the "CONCAT(accesses_date, ' ', accesses_time)" is missing an ")" in the end.

Course I know that I could (and probably should) build a view instead, but I think it would be helpful.

Best,
Bruno

**Comment by torkelo** [comment]
This has already been reported, there is an issue for it also, on mobile now so can't look it up, but you should be able to find it if you search for paranthesis within macro

**Comment by brunomagalhaes** [comment]
Is this it? https://github.com/grafana/grafana/issues/8411

**Comment by torkelo** [comment]
yes

**Comment by Purna17** [comment]
Hi Torkelo,

Is this issue fixed in grafana 5.0.0;

As i'm facing the same issue in that version.. Can you share the work around.

Grafana Genearted query as below it results Null values... Source is Mysql

SELECT sum(LY.LY_Sales) FROM
(SELECT UNIX_TIMESTAMP(DATE_ADD(fiscal_date, INTERVAL 12 HOUR)) as FISCAL_DATE,
sum(Net_Sales_Retail_LY_Shifted ) as LY_Sales
FROM grafana.flashsales_ly
group by 1
order by 1
) LY where FISCAL_DATE >= 1537813800 AND FISCAL_DATE <= 1537867389
